#!/usr/bin/env python3
# _*_ coding:utf-8 _*_

PACKAGE = "rm_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("g_nms_thresh",double_t, 1, "g_nms_thresh", 0.1,0.1, 1.0)
gen.add("g_bbox_conf_thresh",double_t, 2, "g_bbox_conf_thresh", 0.1,0.1, 1.0)
gen.add("g_turn_on_image",bool_t,3,"g_turn_on_image",True)
gen.add("target_is_red",bool_t,5,"target_is_red",True)
gen.add("target_is_blue",bool_t,6,"target_is_blue",False)
gen.add("ratio_of_pixels",double_t, 7, "ratio_of_pixels", 0.10,0.01, 1.00)
gen.add("pixels_thresh",int_t, 8, "pixels_thresh", 0,0, 255)
gen.add("binary_threshold",int_t, 9, "binary_threshold", 100,0, 255)
gen.add("aspect_ratio",double_t, 9, "aspect_ratio", 0.50,0, 2.00)


exit(gen.generate(PACKAGE, PACKAGE, "dynamic"))
u